src/main/java/com/systelab/seed/infrastructure/auth/TokenAuthenticationFilter.java

```diff
@@ -41,53 +37,48 @@
     private ResourceInfo resourceInfo;
 
     @Override
-    public void filter(ContainerRequestContext requestContext) throws IOException {
-        String userRole = "";
+    public void filter(ContainerRequestContext requestContext) {
 
         logger.info(requestContext.getUriInfo().getRequestUri().toString());
-        String authorizationHeader = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
 
-        if (authorizationHeader == null || !authorizationHeader.startsWith("Bearer ")) {
-            logger.severe("Invalid authorizationHeader : " + authorizationHeader);
-            throw new NotAuthorizedException("Authorization header must be provided");
+        String token = getTokenFromHeader(requestContext);
+        if (token != null) {
+            try {
+                String userRole = tokenGenerator.getRoleFromToken(token);
+                if (!methodIsAllowed(resourceInfo.getResourceMethod(), userRole)) {
+                    requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).build());
+                }
+            } catch (Exception ex) {
+                logger.log(Level.SEVERE, "Invalid token : " + token, ex);
+                requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).build());
+            }
+        } else {
+            logger.severe("Invalid authorizationHeader");
+            throw new NotAuthorizedException("Valid authorization header must be provided");
         }
+    }
 
-        String token = authorizationHeader.substring("Bearer".length()).trim();
+    private String getTokenFromHeader(ContainerRequestContext requestContext) {
+        String authorizationHeader = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
 
-        try {
-            userRole = tokenGenerator.validateToken(token);
-        } catch (Exception ex) {
-            logger.log(Level.SEVERE, "Invalid token : " + token, ex);
-            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).build());
+        if (authorizationHeader != null && authorizationHeader.startsWith("Bearer ")) {
+            return authorizationHeader.substring("Bearer".length()).trim();
```

Comment:
Could we have a general variable instead of a hardcoded `String`?

Reply:
Done